### PR TITLE
Newhint#21

### PR DESCRIPTION
**Pull request by mfachal**
# Add hint #21

## Description

Add hint #21: SQUARE_SLOPE_X_MOD_P

## Checklist
- [ ] Linked to Github Issue
- [x] Unit tests added
- [X] Integration tests added.
- [X] This change requires new documentation.
  - [X] Documentation has been added/updated.
  - [X] CHANGELOG has been updated.

### CODE REVIEW

File: src/hint_processor/builtin_hint_processor/secp/ec_utils.rs [hunk 1]
```diff
@@ -226,6 +226,46 @@ pub fn compute_slope(
...
+    let x0 = point0.x.pack86();
+    let x1 = point1.x.pack86();
+    let y0 = point0.y.pack86();
+
+    let value = (slope.pow(2) - &x0 - &x1).mod_floor(&secp_p);
+
+    exec_scopes.insert_value("slope", slope);
+    exec_scopes.insert_value("x0", x0);
+    exec_scopes.insert_value("x1", x1);
+    exec_scopes.insert_value("y0", y0);
```
Comment (fmoletta) [inline src/hint_processor/builtin_hint_processor/secp/ec_utils.rs:262]: Are these values used by other hints in the lib? If not we can just skip adding them to the scope

**Review by fmoletta — APPROVED** [review]
Looks good! Left a small comment

**Review by pefontana — COMMENTED** [review]
Nice one @mfachal 
Also, can you add a unit test?

File: src/hint_processor/builtin_hint_processor/secp/ec_utils.rs (hunk 1 above)
Comment (Oppen) [reply src/hint_processor/builtin_hint_processor/secp/ec_utils.rs:262]: I think I suggested adding them for completion/compatibility.

**Review by pefontana — APPROVED** [review]

**Pull request merged**